### PR TITLE
Update to tested make target for java grpc client.

### PR DESCRIPTION
**Pull request by merlante**
## Describe your changes

- Update to the tested make target to generate the java files for the java grpc client.

## Ticket reference (if applicable)
Fixes #

## Checklist

* [x] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?

* [x] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [x] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [x] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [x] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,16 +47,20 @@ api:
...
-    --openapi_out=fq_schema_naming=true,default_response=false:. \
-    $(API_PROTO_FILES)
+.PHONY: java_api
+# generate java_api proto
+# REQUIRED: protoc-gen-grpc-java plugin from https://repo.maven.apache.org/maven2/io/grpc/protoc-gen-grpc-java/1.63.0/
+# on your path. Then:
+#   chmod u+x <PATH>/protoc-gen-grpc-java-1.63.0-osx-aarch_64.exe
+#   ln -s <PATH>/protoc-gen-grpc-java-1.63.0-<ARCH>.exe protoc-gen-grpc-java
+java_api:
+	mkdir -p ./relations-client-java/src/main/java
```
Comment (Rajagopalan-Ranganathan) [inline Makefile:57]: What happens if the directory exists already?

Comment (merlante) [reply Makefile:57]: [mkdir(1) - Linux man page](https://man7.org/linux/man-pages/man1/mkdir.1.html)
mkdir - make directories
`-p, --parents
              no error if existing, make parent directories as needed,
              with their file modes unaffected by any -m option.`

So if the direction already exists, it does nothing.

Comment (Rajagopalan-Ranganathan) [reply Makefile:57]: Perfect!

**Comment by Rajagopalan-Ranganathan** [comment]
LGTM

**Review by Rajagopalan-Ranganathan — APPROVED** [review]

**Pull request merged**